ksql-common/src/test/java/io/confluent/ksql/util/SchemaUtilTest.java

```diff
@@ -44,4 +63,158 @@ public void shouldCreateCorrectAvroSchema() throws IOException {
+    Class class5 = SchemaUtil.getJavaType(schema5);
+    Class class6 = SchemaUtil.getJavaType(schema6);
+    Class class7 = SchemaUtil.getJavaType(schema7);
+
+    assertThat(class1, equalTo(Boolean.class));
+    assertThat(class2, equalTo(Integer.class));
+    assertThat(class3, equalTo(Long.class));
+    assertThat(class4, equalTo(Double.class));
+    assertThat(class5, equalTo(String.class));
+    assertThat(class6, equalTo(new Double[]{}.getClass()));
+    assertThat(class7, equalTo(HashMap.class));
+
+    try {
+      Class class8 = SchemaUtil.getJavaType(Schema.BYTES_SCHEMA);
+    } catch (KsqlException ksqlException) {
+      Assert.assertTrue(ksqlException.getMessage().equals("Type is not supported: BYTES"));
+      return;
+    }
+    Assert.fail();
+  }
+
+  @Test
+  public void shouldGetTheCorrectFieldName() {
+    Optional<Field> field = SchemaUtil.getFieldByName(schema, "orderid".toUpperCase());
+    Assert.assertTrue(field.isPresent());
+    Assert.assertTrue(field.get().schema() == Schema.INT64_SCHEMA);
+    Assert.assertTrue(field.get().name().equalsIgnoreCase("orderid"));
+
+    Optional<Field> field1 = SchemaUtil.getFieldByName(schema, "orderid");
+    Assert.assertFalse(field1.isPresent());
+  }
+
+  @Test
+  public void shouldGetTheCorrectSchema() {
+    Schema schema1 = SchemaUtil.getTypeSchema("VARCHAR");
+    Schema schema2 = SchemaUtil.getTypeSchema("INT");
+    Schema schema3 = SchemaUtil.getTypeSchema("BIGINT");
+    Schema schema4 = SchemaUtil.getTypeSchema("DOUBLE");
+    Schema schema5 = SchemaUtil.getTypeSchema("BOOLEAN");
+    Schema schema6 = SchemaUtil.getTypeSchema("ARRAY<DOUBLE>");
+    Schema schema7 = SchemaUtil.getTypeSchema("MAP<VARCHAR, DOUBLE>");
+
+    Assert.assertTrue(schema1 == Schema.STRING_SCHEMA);
+    Assert.assertTrue(schema2 == Schema.INT32_SCHEMA);
+    Assert.assertTrue(schema3 == Schema.INT64_SCHEMA);
+    Assert.assertTrue(schema4 == Schema.FLOAT64_SCHEMA);
+    Assert.assertTrue(schema5 == Schema.BOOLEAN_SCHEMA);
+    Assert.assertTrue(schema6.type() == Schema.Type.ARRAY);
+    Assert.assertTrue(schema7.type() == Schema.Type.MAP);
+
+    try {
+      Schema schema8 = SchemaUtil.getTypeSchema("BYTES");
+    } catch (Exception e) {
+      Assert.assertTrue(e.getMessage().equals("Unsupported type: BYTES"));
+      return;
+    }
+    Assert.fail();
+  }
+
+  @Test
+  public void shouldGetTheCorrectFieldIndex() {
+    int index1 = SchemaUtil.getFieldIndexByName(schema, "orderid".toUpperCase());
+    int index2 = SchemaUtil.getFieldIndexByName(schema, "itemid".toUpperCase());
+    int index3 = SchemaUtil.getFieldIndexByName(schema, "mapcol".toUpperCase());
+    int index4 = SchemaUtil.getFieldIndexByName(schema, "mapcol1".toUpperCase());
+
+    assertThat("Incorrect index.", index1, equalTo(1));
+    assertThat("Incorrect index.", index2, equalTo(2));
+    assertThat("Incorrect index.", index3, equalTo(5));
+    assertThat("Incorrect index.", index4, equalTo(-1));
```

Comment:
what does this one mean?
Seems like it should be a different test to me

Reply:
Moved the incorrect field name test to a new test.